### DISCUSSION TITLE
Attaching a color texture to an existing render target?

### DISCUSSION THREAD

**Discussion by yavuz650** [google/filament#6400] [Q&A] [answered]
I attach the color attachment to a render target as shown in the `rendertarget` sample. This only allows me to attach a color texture while building the render target. How can I attach another color texture to an existing render target? 

Maybe something like, `rendertarget.setTexture(COLOR, newColorTexture)`

**Comment by bejado** [comment, marked as answer]
`RenderTarget`s are immutable, so you'll need to create two `RenderTarget`s, one for each color texture.

**Reply by yavuz650** [reply]
Is there a reasoning behind this? From what I can tell, `RenderTarget` represents a framebuffer. At least in OpenGL, framebuffers are mutable. 

Just curious.

**Reply by romainguy** [reply]
Immutable data structures are easier to reason about and we support other backends than GL. Mutating framebuffers in GL also comes at a cost so there's not real benefit in allowing mutation anyway.
